app/src/main/java/com/concordia/mcga/utilities/pathfinding/PathFinder.java

```diff
@@ -53,6 +56,41 @@ public PathFinder(TiledMap map) {
         return returnList;
     }
 
+    public List<PathFinderTile> shortestPathJunctions(int startX, int startY, int destX, int destY) throws Exception {
+        ArrayList<PathFinderTile> pathTiles = new ArrayList<PathFinderTile>(shortestPath(startX, startY, destX, destY));
+        ArrayList<PathFinderTile> pathTilesJunctions = new ArrayList<PathFinderTile>();
+
+        int curX = 0;
+        int curY = 0;
+        for (int i = 0; i < pathTiles.size(); i++) {
+            if (i == 0) {
+                PathFinderTile pft = pathTiles.get(i);
+                pathTilesJunctions.add(pft);
+                curX = pft.getCoordinateX();
+                curY = pft.getCoordinateY();
+            } else {
+                PathFinderTile pft = pathTiles.get(i);
+                if (!(pft.getCoordinateX() == curX && pft.getCoordinateY() != curY) && !(pft.getCoordinateY() == curY && pft.getCoordinateX() != curX)) {
+                    pathTilesJunctions.add(pft);
+                    curX = pft.getCoordinateX();
+                    curY = pft.getCoordinateY();
+                }
+            }
+        }
+
+        return pathTilesJunctions;
+    }
+
+    public static JSONArray toJSONArray(List<PathFinderTile> pathTilesJunctions) {
+        JSONArray pftArray = new JSONArray();
+        Iterator<PathFinderTile> it = pathTilesJunctions.iterator();
+        while (it.hasNext()) {
```

Comment:
It's a personal preference. I use iterators most of the time for lists because iterators are thread safe and never result in ConcurrentModificationException. There is not performance drawback.